src/utils/eventTable.tsx

```diff
@@ -0,0 +1,116 @@
+import { LinkWithQuery } from "@/components/LinkWithQuery";
+import { IInteractiveTableColumn, IInteractiveTableRow } from "@/components/organisms/interactiveTable/InteractiveTable";
+import { TFamilyConcept, TFamilyDocumentPublic, TFamilyEventPublic, TFamilyPublic } from "@/types";
+
+import { formatDateShort } from "./timedate";
+
+/* Columns */
+
+export type TEventTableColumnId = "action" | "caseNumber" | "caseTitle" | "court" | "date" | "document" | "matches" | "summary" | "type";
+export type TEventTableColumn = IInteractiveTableColumn<TEventTableColumnId>;
+
+export const getEventTableColumns = ({
+  isUSA = true,
+  showFamilyColumns = false,
+  showMatches = false,
+}: {
+  isUSA?: boolean;
+  showFamilyColumns?: boolean;
+  showMatches?: boolean;
+}) => {
+  let columns: TEventTableColumn[] = [
+    { id: "date", name: "Filing Date", fraction: 2 },
+    { id: "type", fraction: 3 },
+    { id: "action", name: "Action taken", fraction: 3 },
+    { id: "document" },
+    { id: "summary", sortable: false, fraction: 6 },
+  ];
+
+  if (showFamilyColumns) {
+    columns = [...columns, { id: "caseNumber", name: "Case", fraction: 2 }, { id: "court" }, { id: "caseTitle", name: "Case", fraction: 2 }];
+  }
+
+  if (!isUSA) columns.splice(2, 1);
+  if (showMatches) columns.push({ id: "matches" });
+
+  return columns;
+};
+
+/* Rows */
+
+export type TEventTableRow = IInteractiveTableRow<TEventTableColumnId>;
+
+type TEventWithDocument = {
+  event: TFamilyEventPublic;
+  document?: TFamilyDocumentPublic;
+};
+
+const getMostSpecificCourt = (concepts: TFamilyConcept[]): TFamilyConcept | null => {
+  let courtConcepts = concepts.filter((concept) => concept.type === "legal_entity");
+  if (courtConcepts.length === 0) return null;
+
+  // On each loop, remove legal entities without parents. Stops when the deepest level court remains
+  while (courtConcepts.length > 1) {
+    const moreSpecificConcepts = courtConcepts.filter((concept) =>
+      concept.subconcept_of_labels.some((id) => concepts.findIndex((con) => con.id === id) !== -1)
+    );
+
+    // Prevent a situation where number of concepts goes from 2 to 0 on the last loop
+    courtConcepts = moreSpecificConcepts.length === 0 ? courtConcepts.slice(0, 1) : moreSpecificConcepts;
```

Comment:
Pondering: in situations where there are two lowest level courts, should we display both rather than the first instance?